torch/_dynamo/variables/tensor.py

```diff
@@ -720,6 +722,55 @@ def call_method(
     ) -> "VariableTracker":
         unimplemented(f"numpy_ndarray.{name}()")
 
+    @staticmethod
+    def reconstruct_ndarray_before_return(codegen, output_graph) -> List[Instruction]:
```

Comment:
You might be able to simplify this by defining:
```
def to_numpy_helper(___tmp_0):
    if isinstance(___tmp_0, torch_np.ndarray):
        return ___tmp_0.tensor.numpy()
    else:
        return ___tmp_0
```

Then calling it similar to:
https://github.com/pytorch/pytorch/blob/9244264f469036ddd8868cece96f640bdbd2ccfc/torch/_dynamo/side_effects.py#L342